crates/relayer/src/link/relay_path.rs

```diff
@@ -1381,11 +1389,55 @@ impl<ChainA: ChainHandle, ChainB: ChainHandle> RelayPath<ChainA, ChainB> {
         dst_info: &ChainStatus,
         height: Height,
     ) -> Result<(Option<Any>, Option<Any>), LinkError> {
+        crate::time!(
+            "build_recv_or_timeout_from_send_packet_event",
+            {
+                "src_channel": event.packet.source_channel,
+                "dst_channel": event.packet.destination_channel,
+            }
+        );
         let timeout = self.build_timeout_from_send_packet_event(event, dst_info)?;
         if timeout.is_some() {
             Ok((None, timeout))
         } else {
-            Ok((self.build_recv_packet(&event.packet, height)?, None))
+            match event
+                .packet
+                .are_fields_valid(self.max_memo_size, self.max_receiver_size)
+            {
+                ValidationResult::Valid => {
+                    Ok((self.build_recv_packet(&event.packet, height)?, None))
+                }
+                ValidationResult::MemoTooBig { size, max } => {
+                    trace!("packet: {:#?}", event.packet);
+                    debug!("memo field of packet with sequence {} is too big. Memo field size {size}, maximum configured size {max}", event.packet.sequence);
+                    Ok((None, None))
+                }
+                ValidationResult::ReceiverTooBig { size, max } => {
+                    trace!("packet: {:#?}", event.packet);
+                    debug!("receiver field of packet with sequence {} is too big. Receiver field size {size}, maximum configured size {max}", event.packet.sequence);
+                    Ok((None, None))
+                }
+                ValidationResult::BothTooBig {
+                    memo_size,
+                    memo_max,
+                    receiver_size,
+                    receiver_max,
+                } => {
+                    trace!("packet: {:#?}", event.packet);
+                    debug!(
+                        "memo and receiver field of packet with sequence {} are too big. \
+                        Memo field size {memo_size}, maximum configured size {memo_max}, \
+                        receiver field size {receiver_size}, maximum configured size {receiver_max}",
+                        event.packet.sequence
+                    );
+                    Ok((None, None))
+                }
+                ValidationResult::DecodeFailure { details } => {
+                    trace!("packet: {:#?}", event.packet);
+                    trace!("failed to decode packet or packet isn't an ICS20 packet, details: {details}");
```

Comment:
Is there a way to determine whether or not the packet was supposed be a valid ICS-20 packet?